### PR TITLE
refactor: migrate react-script in the sso-tab template (TS & JS) to vite

### PR DESCRIPTION
**Pull request by Yimin-Jin**
BUG: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/29050459

### CODE REVIEW

**Review by hund030 — APPROVED** [review]

File: templates/js/sso-tab-with-obo-flow/package.json.tpl [hunk 1]
```diff
@@ -13,25 +14,28 @@
...
   },
   "devDependencies": {
     "@types/node": "^18.0.0",
     "@types/react": "^18.0.0",
     "@types/react-dom": "^18.0.0",
     "@types/react-router-dom": "^5.3.3",
     "concurrently": "^8.2.2",
-    "env-cmd": "^10.1.0"
+    "env-cmd": "^10.1.0",
+    "typescript": "^4.1.2",
```
Comment (huimiu) [inline templates/js/sso-tab-with-obo-flow/package.json.tpl:26]: We don't need the typescript dependency in a JS project.

Comment (Yimin-Jin) [reply templates/js/sso-tab-with-obo-flow/package.json.tpl:26]: Right, I'll remove it.

File: templates/ts/sso-tab-with-obo-flow/tsconfig.json [hunk 2]
```diff
@@ -14,7 +14,8 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react-jsx"
+    "jsx": "react-jsx",
```
[the remark this answers is not in the document]
Comment (huimiu) [reply templates/ts/sso-tab-with-obo-flow/tsconfig.json:17]: https://stackblitz.com/edit/vitejs-vite-5mgrx2?file=index.html&terminal=dev

[the remark this answers is not in the document]
Comment (Yimin-Jin) [reply templates/ts/sso-tab-with-obo-flow/tsconfig.json:17]: I don't think this change is necessary. According to the official document, we should replace all of `process.env` with `import.meta.env`, and also change all `REACT_` to `VITE_` . I think we should aim for the minimal required changes.

[the remark this answers is not in the document]
Comment (Yimin-Jin) [reply templates/ts/sso-tab-with-obo-flow/tsconfig.json:17]: I am following  this official document, which directly uses `tsconfig.json` and does not mention the need to differentiate between Vite and Node.
[https://vitejs.dev/guide/features.html](url)

**Review by huimiu — APPROVED** [review]

**Pull request merged**